include/socket_offload_ops.h

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2017 Texas Instruments
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Socket Offload Redirect API
+ */
+
+#ifndef __SOCKET_OFFLOAD_OPS_H__
+#define __SOCKET_OFFLOAD_OPS_H__
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <sys/types.h>
+#include <net/net_ip.h>
+
+/*
+ * It is assumed that these offload functions (except for init()) follow the
+ * POSIX socket API standard for arguments, return values and setting of errno.
+ */
+struct socket_offload {
```

Comment:
Why this structure goes into a separate header, why we can't have single socket_offload.h?

Reply:
"Principle of minimum inclusion".  The idea is to only include what's needed.  The socket provider only needs the definitions in socket_offload_ops.h, but not the definitions in socket_offload.h.